tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -590,6 +590,32 @@ public:
    std::string GetActionName() { return "Take"; }
 };
 
+// Extern templates for TakeHelper
+// NOTE: The move-constructor of specializations declared as extern templates
+// must be defined out of line, otherwise cling fails to find its symbol.
+template <typename RealT_t, typename T, typename COLL>
+TakeHelper<RealT_t, T, COLL>::TakeHelper(TakeHelper<RealT_t, T, COLL> &&) = default;
+template <typename RealT_t, typename T>
+TakeHelper<RealT_t, T, std::vector<T>>::TakeHelper(TakeHelper<RealT_t, T, std::vector<T>> &&) = default;
+template <typename RealT_t, typename COLL>
+TakeHelper<RealT_t, RVec<RealT_t>, COLL>::TakeHelper(TakeHelper<RealT_t, RVec<RealT_t>, COLL> &&) = default;
+template <typename RealT_t>
+TakeHelper<RealT_t, RVec<RealT_t>, std::vector<RealT_t>>::TakeHelper(TakeHelper<RealT_t, RVec<RealT_t>, std::vector<RealT_t>> &&) = default;
+
+// std::vector<bool> does not have an emplace_back method before cxx14
+#if __cplusplus < 201400L
+extern template class TakeHelper<bool, bool, std::vector<bool>>;
+#endif
```

Comment:
this should not be needed. If there is a bug in `TakeHelper<bool>` we should fix :smile: